src/clr-angular/data/datagrid/datagrid-hideable-column.spec.ts

```diff
@@ -56,6 +56,34 @@ export default function(): void {
         expect(context.clarityDirective.columnId).toEqual(context.clarityDirective.hideable.id);
       });
     });
+
+    describe('TypeScript Output API', function() {
+      let context: TestContext<ClrDatagridColumn<void>, HideableOutputTest>;
+
+      beforeEach(function() {
+        context = this.create(ClrDatagridColumn, HideableTest, PROVIDERS_NEEDED);
+      });
+
+      it('creates a DatagridHideableColumn instance on the DatagridColumn', function() {
+        expect(context.clarityDirective.hideable).toBeDefined();
+      });
+
+      it('defaults the HideableColumn.hidden property to false', function() {
+        expect(context.clarityDirective.hideable.hidden).toBe(false);
+      });
+
+      it('correctly populates the DatagridHideableColumn instance with an id', function() {
+        expect(context.clarityDirective.columnId).toEqual(context.clarityDirective.hideable.id);
+      });
+
+      it('emits column state change', function() {
+        let hiddenState = false;
+        context.testComponent.directive.hiddenChange.subscribe(state => (hiddenState = state));
```

Comment:
Since you're using the two-way binding syntax in your test component, testing that the output emits correctly is simply testing that your test component's value is up to date. This means you don't need a subscription:
```typescript
it('emits column state change', function() {
  context.clarityDirective.hideable.hidden = true;
  expect(context.testComponent.hidden).toBeTrue();
});
```

Reply:
you're completely right, could've though of this on my own. thanks! fixed.